### PR TITLE
add shell formatter to formatter docker image

### PR DESCRIPTION
**Pull request by kmcnellis**
### Description
- Add the shfmt and shellchecker bash script linters to the formatter docker image and github actions
- Format the files so that the tests pass

### CODE REVIEW

File: .github/workflows/format.yaml [hunk 1]
```diff
@@ -38,10 +38,19 @@ jobs:
...
         run: terraform fmt -check -recursive -diff cloud
+  sh-checker:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Run the sh-checker
+        uses: luizm/action-sh-checker@master
+        env:
+          SHELLCHECK_OPTS: -s bash -S error
+          SHFMT_OPTS:  -bn -ci -kp -i 2 # Google Bash Style
```
Comment (clouser2) [inline .github/workflows/format.yaml:56]: Non-Blocking: For future readers, would you mind adding a link that explains each option. Or if there was a recipe that showed the options for Google Bash Style, link it instead?

File: bin/azurite/wait [hunk 2]
```diff
@@ -23,7 +23,7 @@ healthy() {
 }
 
 until healthy "$(curl --silent --max-time 2 "${azurite_endpoint}/")"; do
-  if (( "$(date +%s)" > "${DEADLINE}" )); then
```
Comment (clouser2) [inline bin/azurite/wait:26]: Non-blocking: Out of curiosity, is it reasonable to assume that all of the shell file changes here were purely the result of running the formatter and didn't require any manual changes on your part.

File: universal-application-tool-0.0.1/.settings/org.eclipse.core.resources.prefs [hunk 3]
```diff
@@ -0,0 +1,3 @@
+#Generated by sbteclipse
```
Comment (clouser2) [inline universal-application-tool-0.0.1/.settings/org.eclipse.core.resources.prefs:1]: Is this inclusion accidental or is it required to make the formatter work?

**Review by clouser2 — APPROVED** [review]

File: .gitignore [hunk 4]
```diff
@@ -40,3 +41,6 @@ sbt_cache/*
 **/.ammonite/
 **/metals.sbt
 **/pom.xml
+universal-application-tool-0.0.1/package-lock.json
+universal-application-tool-0.0.1/pom.bkp
+universal-application-tool-0.0.1/.sbtopts
```
Comment (bion) [inline .gitignore:46]: issue (blocking): these gitignore changes are unrelated to the rest of the PR, would prefer to have it separate

File: .gitignore [hunk 5]
```diff
@@ -40,3 +41,6 @@ sbt_cache/*
 **/.ammonite/
 **/metals.sbt
 **/pom.xml
+universal-application-tool-0.0.1/package-lock.json
```
Comment (bion) [inline .gitignore:44]: issue (blocking): why ignore the package lock file? without it package versions will change without us noticing

File: .scalafmt.conf [hunk 6]
```diff
@@ -0,0 +1,2 @@
+version = "3.4.0"
+runner.dialect = scala213
```
Comment (bion) [inline .scalafmt.conf:2]: question (blocking): what's this for?

File: .shellcheckrc [hunk 7]
```diff
@@ -0,0 +1,21 @@
+# Allow opening any 'source'd file, even if not specified as input
```
Comment (bion) [inline .shellcheckrc:1]: suggestion (blocking): please add some comments to this file explaining what it is / what uses it

File: bin/azurite/wait [hunk 8]
```diff
@@ -23,7 +23,7 @@ healthy() {
...
+  if (("$( date +%s)" > "${DEADLINE}")); then
```
Comment (bion) [inline bin/azurite/wait:26]: question (non-blocking): the space before `date` here looks a little funny to me, did the formatter do this to get it to match the style guide?

File: cloud/azure/bin/setup-saml-keystore [hunk 9]
```diff
@@ -39,8 +38,8 @@ if [[ ! "${resource_group}" ]] \
   exit 1
 fi
 
-if key_vault::has_secret "${vault_name}" "${KEYSTORE_PASS_SECRET_NAME}" &&
-  storage::has_blob "${storage_account_name}" \
+if key_vault::has_secret "${vault_name}" "${KEYSTORE_PASS_SECRET_NAME}" \
+                                                                        && storage::has_blob "${storage_account_name}" \
```
Comment (bion) [inline cloud/azure/bin/setup-saml-keystore:42]: question (blocking): is there a setting we can change that won't result in such deep indentation for continuations?

File: cloud/shared/pybin/doctor_impl.py [hunk 10]
```diff
@@ -1,5 +1,3 @@
-#! /usr/bin/env python3
```
Comment (bion) [inline cloud/shared/pybin/doctor_impl.py:1]: question (blocking): why move this to a different directory?

File: universal-application-tool-0.0.1/.prettierrc.js [hunk 11]
```diff
@@ -0,0 +1,3 @@
+module.exports = {
+  ...require('gts/.prettierrc.json'),
+}
```
Comment (bion) [inline universal-application-tool-0.0.1/.prettierrc.js:3]: suggestion (blocking): happy to have more autoformatting for JS/TS! would prefer it to be in a separate PR though

**Review by bion — CHANGES_REQUESTED** [review]

File: bin/azurite/wait (hunk 8 above)
Comment (kmcnellis) [reply bin/azurite/wait:26]: Hmmm

Comment (kmcnellis) [reply bin/azurite/wait]: OK, it's done now.

Comment (kmcnellis) [reply bin/azurite/wait]: (It was a bug in the -kp flag : https://github.com/mvdan/sh/issues/362)

**Review by bion — APPROVED** [review]
